spectral_cube/spectral_cube.py

```diff
@@ -340,11 +340,64 @@ def __repr__(self):
             s += ":\n"
         else:
             s += " and unit={0}:\n".format(self.unit)
-        s += " n_x: {0}  type_x: {1:8s}  unit_x: {2}\n".format(self.shape[2], self.wcs.wcs.ctype[0], self.wcs.wcs.cunit[0])
-        s += " n_y: {0}  type_y: {1:8s}  unit_y: {2}\n".format(self.shape[1], self.wcs.wcs.ctype[1], self.wcs.wcs.cunit[1])
-        s += " n_s: {0}  type_s: {1:8s}  unit_s: {2}".format(self.shape[0], self.wcs.wcs.ctype[2], self.wcs.wcs.cunit[2])
+        s += (" n_x: {0:6d}  type_x: {1:8s}  unit_x: {2:5s}"
+              "  range: {3:8s}:{4:8s}\n".format(self.shape[2],
+                                                self.wcs.wcs.ctype[0],
+                                                self.wcs.wcs.cunit[0],
+                                                self.longitude_extrema[0],
+                                                self.longitude_extrema[1],
+                                               ))
+        s += (" n_y: {0:6d}  type_y: {1:8s}  unit_y: {2:5s}"
+              "  range: {3:8s}:{4:8s}\n".format(self.shape[1],
+                                                self.wcs.wcs.ctype[1],
+                                                self.wcs.wcs.cunit[1],
+                                                self.latitude_extrema[0],
+                                                self.latitude_extrema[1],
+                                               ))
+        s += (" n_s: {0:6d}  type_s: {1:8s}  unit_s: {2:5s}"
+              "  range: {3:8s}:{4:8s}".format(self.shape[0],
+                                              self.wcs.wcs.ctype[2],
+                                              self.wcs.wcs.cunit[2],
+                                              self.spectral_extrema[0],
+                                              self.spectral_extrema[1],
+                                             ))
         return s
 
+    @property
```

Comment:
How do you feel about making a `cached_property` decorator, instead of checking for all these `hasattr` checks?